client/src/main/java/io/pravega/client/control/impl/ModelHelper.java

```diff
@@ -263,20 +263,19 @@ public static final SegmentWithRange encode(final SegmentRange segmentRange) {
      * @return map of segment to position
      */
     public static ReaderGroupConfig encode(Controller.ReaderGroupConfiguration rgConfig) {
-        return ReaderGroupConfig.builder()
+        ReaderGroupConfig cfg = ReaderGroupConfig.builder()
                 .automaticCheckpointIntervalMillis(rgConfig.getAutomaticCheckpointIntervalMillis())
                 .groupRefreshTimeMillis(rgConfig.getGroupRefreshTimeMillis())
                 .maxOutstandingCheckpointRequest(rgConfig.getMaxOutstandingCheckpointRequest())
                 .retentionType(ReaderGroupConfig.StreamDataRetention.values()[rgConfig.getRetentionType()])
-                .generation(rgConfig.getGeneration())
-                .readerGroupId(UUID.fromString(rgConfig.getReaderGroupId()))
                 .startingStreamCuts(rgConfig.getStartingStreamCutsList().stream()
                         .collect(Collectors.toMap(streamCut -> Stream.of(streamCut.getStreamInfo().getScope(), streamCut.getStreamInfo().getStream()),
                                 streamCut -> generateStreamCut(streamCut.getStreamInfo().getScope(), streamCut.getStreamInfo().getStream(), streamCut.getCutMap()))))
                 .endingStreamCuts(rgConfig.getEndingStreamCutsList().stream()
                         .collect(Collectors.toMap(streamCut -> Stream.of(streamCut.getStreamInfo().getScope(), streamCut.getStreamInfo().getStream()),
                                 streamCut -> generateStreamCut(streamCut.getStreamInfo().getScope(), streamCut.getStreamInfo().getStream(), streamCut.getCutMap()))))
                 .build();
+        return ReaderGroupConfig.cloneConfig(cfg, UUID.fromString(rgConfig.getReaderGroupId()), rgConfig.getGeneration());
```

Comment:
if the protobuf message is from an older client, then readergroupId could be null.. but older client will never call any of the new RG apis as they are only added with new api. so i suppose its fine.

Reply:
If message if from olderClient, ReaderGroupId will be DEFAULT_UUID and not null.
On the Controller, in the createReaderGroupAPI() I have added a validation check where if readerGroupId is DEFAULT_UUID, I replace it with `UUID.randomUUID`  as we'd never want a ReaderGroup to be created with DEFAULT_ID

Reply:
this api is never called from older client though :). so it will always have correct UUID.